db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,124 @@
+      WHERE purged = FALSE
+    SQL
+
+    # automatic_translations will take the format of { 'English' : 'eng' }
+    # Where `English` is plucked directly from the DB
+    automatic_translations = {}
+
+    # custom_translations will take the format of { 'portugese' : 'eng' }
+    # Where `portugese` is manually defined as a value that was manually selected from the DB (from manual_translations)
+    # The only different between `custom_translations` and `automatic_translations` is that
+    # when languages are updated with custom_translations, the Patients will have a HistoryItem
+    # added to them. Patients who are translated with automatic_translations will NOT have that item on them
+    custom_translations = {}
+
+    # Update this hash as necessary with any manual translations/misspellings
+    # The 'key' part can be display name, or iso code. It will be correctly translated before any execution happens
+    manual_translations = {
+      'bilingual english/spanish': 'english',
+      'portugese': 'portuguese',
+      'bos': 'bosnian',
+      'laotian': 'lao',
+      'pelugu': 'telugu',
+      'telegu': 'telugu',
+      'philippines': 'filipino',
+      'punjabi': 'panjabi',
+      'frysian': 'frisian',
+      'haitian creole': 'haitian',
+    }
+    # Get one master hash of languages to iso codes
+    manual_translations.each do |key, value|
+      custom_translations[key.to_sym] = (match_language(value) || 'eng').to_s
+    end
+
+    existing_primary_languages = Patient.where(purged: false).distinct.pluck(:primary_language)
+    existing_primary_languages.reject!(&:blank?)
+    existing_secondary_languages = Patient.where(purged: false).distinct.pluck(:secondary_language)
+    existing_secondary_languages.reject!(&:blank?)
+
+    # Create an translation in automatic_translations, unless we already have it defined in `manual_translations`
+    existing_primary_languages.each do |el|
+      automatic_translations[el.to_sym] = (match_language(el) || 'eng').to_s if (manual_translations[el.to_sym].nil?)
+    end
+    # Create an translation in automatic_translations, unless we already have it defined in `manual_translations`
+    existing_secondary_languages.each do |el|
+      automatic_translations[el.to_sym] = (match_language(el) || 'eng').to_s if (manual_translations[el.to_sym].nil?)
+    end
+
+    automatic_translations.each do |key, value|
+      # Patient records updated here are solely updated (no History Items included)
+      Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+      Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+    end
+    custom_translations.each do |key, value|
+      # Patient records updated here WILL have History Items included for them.
+      ids = Patient.where(purged: false).where(primary_language: key.to_s).distinct.pluck(:id)
```

Comment:
I think we should wrap the Patient update and History create in a transaction for data consistency

Reply:
done.